src/components/dashboard/Tile.tsx

```diff
@@ -1,31 +1,122 @@
-import React from "react";
-import { Box, Heading, LinkBox, LinkOverlay, Text } from "@chakra-ui/react";
+import React, { MutableRefObject, useEffect } from "react";
+import {
+  Box,
+  Heading,
+  LinkBox,
+  LinkOverlay,
+  Text,
+  Tag,
+  Flex,
+  AlertDialog,
+  AlertDialogBody,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogContent,
+  AlertDialogOverlay,
+  useDisclosure,
+  Button,
+} from "@chakra-ui/react";
 
 interface Props {
   title: string;
   description: string;
+  branchId: string;
   image?: string;
+  status: string;
+  hasApplication: boolean;
+  currAppLink: string;
+  chooseBranch: (appBranchID: any) => Promise<string | undefined>;
 }
 
-const Tile: React.FC<Props> = props => (
-  <LinkBox borderRadius="4px" boxShadow="rgba(0, 0, 0, 0.24) 0px 3px 8px">
-    <Box
-      bgGradient={props.image ? "" : "linear(to-l, #33c2ff, #7b69ec)"}
-      borderTopRadius="4px"
-      height="150px"
-      onClick={() => console.log("Clicked!")}
-    />
-    <Box padding="20px 32px">
-      <Heading fontSize="18px" fontWeight="semibold" marginBottom="10px" color="#212121">
-        <LinkOverlay href="#">
-          {props.title}
-        </LinkOverlay>
-      </Heading>
-      <Text fontSize="sm" color="#858585">
-        {props.description}
-      </Text>
-    </Box>
-  </LinkBox>
-);
+
+const Tile: React.FC<Props> = props => {
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const cancelRef = React.useRef(null)
+
+  const onConfirm = async () => {
+    onClose()
+    try {
+      const newAppLink = await props.chooseBranch(props.branchId)
+      window.open(newAppLink)
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  return (
+    <>
+      <AlertDialog
+        isOpen={isOpen}
+        leastDestructiveRef={cancelRef}
+        onClose={onClose}
+      >
+        <AlertDialogOverlay>
+          <AlertDialogContent>
+            <AlertDialogHeader fontSize='lg' fontWeight='bold'>
+              Start a new application?
+            </AlertDialogHeader>
+
+            <AlertDialogBody>
+              Are you sure? You have already started another application. If you start a new one, all previous progress will be lost.
+            </AlertDialogBody>
+
+            <AlertDialogFooter>
+              <Button ref={cancelRef} onClick={onClose}>
+                Cancel
+              </Button>
+              <Button colorScheme='blue' onClick={onConfirm} ml={3}>
+                Confirm
+              </Button>
+            </AlertDialogFooter>
+          </AlertDialogContent>
+        </AlertDialogOverlay>
+      </AlertDialog>
```

Comment:
Right now, there's a weird glitch that when an alert dialog opens up, it scrolls to the top of the page. I think a solution to this would be to move the alert dialog stuff to the parent component. Then, when you click on a branch, just call `props.chooseBranch`. Then, the parent component can determine whether or not to show the popup.

Reply:
I was messing around with this and couldn't get it to work. Put the AlertDialog stuff in Branches that renders all the tiles and it still scrolled back up and seemed to be messier code.